### DISCUSSION TITLE
📣 Take our survey! Provide your feedback on our feedback platform

### DISCUSSION THREAD

**Discussion by jricciardi** [community/community#14965] [Other Feature Feedback, Questions, & Ideas] [answered]
Hello! 👋 

As we’ve been using [GitHub/feedback/discussions](https://github.com/github/feedback/discussions) as a place for product feedback for over a year now, we want to hear from you about what's working and where things can improve. We invite you to participate in this brief 5-minute survey, to help us better understand your experience using this platform. Your feedback will help inform our next iteration of this space, so please let us know what you think! We're excited to hear from you ✨

**[Link to the survey](https://survey3.medallia.com/?VQdmUT-feedback
)**
This survey will close April 28, 2022

Thank you!

**Comment by nersoz-impinj** [comment]
Here is my list of things that are broken in GH code review:

1. I cannot link to a specific code change or comment in a PR.
   There is no "permalink" within a PR by which to reference a topic outside of
   the conversation.

2. There is no concept of "tasks which need to be completed before merge".
   We have enforced "all conversations need resolved" before merge policy.
   But resolving a conversation is a conversation killer. The conversation should
   remain open, and a task of items should be generated from the conversation.
  
  * The task should be linked to the conversation.
  *  When the task list is completed, then the PR can merge.
  * Tasks to complete should be able to be linked via "permalink".

3. The PR file differencing, with the continuous scrolling, with no file
   delineation is terrible. As you're reading through the changes, you suddenly
   realize the context has changed because you passed the end of the file.

   Files provide specific scoping of functions and variables. I am not aware of a
   language that does not provide file based scoping.

   Reviewing of files should be ONE FILE AT A TIME. Not continuous scrolling of text.

4. When changing files, the file in the file tree should be highlighted.
   Not stale, highlighting the file you were reading 5 minutes ago.

5. The notion of clicking the "Viewed" check box is a good idea.
   But when a new commit is pushed and that file changes, the the "Viewed"
   check box should become unchecked.

6. When a comment loses its context, i.e. the associated a file or fragment
   has been renamed, extracted to a new file, force pushed, etc. Then that comment
   needs to be easily found. And it is not.

   One solution:
   Have a list of links to all comments that cannot be lost. Even if its in some
   orphan list of comments, with the file name and line number originally associated.

7. Once conversations are resolved, they should be placed into a different category.
   Here is an idea: "Open conversations and Resolved conversations".

   There is currently no indication of whether there are open conversations or
   resolved conversations. Just "Conversations".

    The only means by which you quickly know whether a new conversation was added to a PR is to memorize the 
    previous conversation count. I actually do this. "Oh, the last conversation count was 24, now it is 27. 
    Now, the PR author, or reviewer, has to scroll through the entire PR searching for "new" conversations.
    This is awful.

8. "Hidden conversations" - What is the point of hiding conversations that have not been resolved? 
    Why are OPEN conversations hidden? WHY?

**Reply by ProphetHyBriD** [reply]
I’m new to github.com & I already can contribute to this by saying: They should add a HELP ME option somewhere on the page where either it is on the top bar and contains FAQ’s, tutorials, (well, at least one good one showing the basics) a list of all the main/major things that you can do on this website and perhaps a separate list within that page/category that illustrates and/or reveals other interesting & non-standard things that have been done or that ppl are doing within this site and maybe an Introduction page in the HELP ME and maybe somewhere else too, like on the homepage…but in that it would start with the goals, any agendas, what uses or blessings have been born from this site…and end it with several paragraphs or whatever telling the audience/world how it all started, who started it, why, when, how, where, AND what the goal is exactly or just whatever! 

So in summary, Add a ‘HELP ME’ button/section (or whatever u wanna call it)

**Reply by palkoi** [reply]
Jók ezek az ötletek és tetszik az oldal

**Reply by palkoi** [reply]
> > مرحبًا!👋
> > نظرا أننا نستخدم Github [/ التعليقات / المناقشات](https://github.com/github/feedback/discussions) كمكان لتعليقات المنتج لأكثر من عام الآن, نريد أن المع منك منك منك من حول ما ينجح وأأشيا يمكن تحسين الأشياء. ندعوك للمشاركة في هذا الاستبيان الموجز الذي يستغرق 5 دقائئ, لمساعدتنا على فهم تجربتك بشكل أفضل باستخدام هذه المنصة. A يسعدنا أن نسمع منك✨
> > **[رابط](https://survey3.medallia.com/?VQdmUT-feedback)** الاستطلاع سيغلق هذا الاستطلاع في 2022. 28. أبريل
> > شكرًا لك!

**Reply by palkoi** [reply]
Mindenkinek segitségre leszek, hogy mindenkinekjobb legyen ebben a közösségben

**Comment by jricciardi** [comment, marked as answer]
Thank you all for participating in our survey! We're closing this today and locking this post.
